### PR TITLE
Update cilium to v1.10.1

### PR DESCRIPTION
**Pull request by DockToFuture**
**How to categorize this PR?**

/area networking
/kind enhancement

**What this PR does / why we need it**:
Update cilium to `v1.10.1`

**Release note**:

```other operator
Update cilium to `v1.10.1`
```

### CODE REVIEW

File: charts/images.yaml [hunk 1]
```diff
@@ -2,15 +2,15 @@ images:
   - name: cilium-agent
     sourceRepository: github.com/cilium/cilium
     repository: quay.io/cilium/cilium
-    tag: v1.9.6
+    tag: v1.10.0
```
Comment (vpnachev) [inline charts/images.yaml:5]: There is v1.10.1, see https://github.com/cilium/cilium/releases/tag/v1.10.1

File: charts/images.yaml [hunk 2]
```diff
@@ -22,20 +22,20 @@ images:
...
   - name: hubble-ui-backend
     sourceRepository: github.com/cilium/hubble-ui-backend
     repository: quay.io/cilium/hubble-ui-backend
-    tag: v0.7.3
+    tag: v0.7.9
   - name: envoy
     sourceRepository: github.com/envoyproxy/envoy
     repository: docker.io/envoyproxy/envoy
-    tag: v1.14.5
+    tag: v1.18.2
```
Comment (vpnachev) [inline charts/images.yaml:33]: There is v1.18.3, see https://github.com/envoyproxy/envoy/releases/tag/v1.18.3

**Review by vpnachev — COMMENTED** [review]
Some components can be upgraded to even newer version :wink:

File: charts/images.yaml (hunk 2 above)
Comment (DockToFuture) [reply charts/images.yaml:33]: There is but is not used in the latest cilium release v1.10.1

**Comment by vpnachev** [comment]
Sorry, I cannot provide valuable review on this change, it is better someone from @gardener/gardener-extension-networking-cilium-maintainers to also take a look and eventually approve it.

**Review by ScheererJ — APPROVED** [review]
/lgtm

**Pull request merged**